docs/user-guide/gaps.md

```diff
@@ -0,0 +1,126 @@
+Understanding gaps between high and low level code
+===================================================
+
+The Certora Prover is analyzing low-level code, such as the EVM bytecode.
+However, the CVL specification as well as the Rule Report and Call Trace are 
+usually presenting information in terms of the high-level language (e.g., Solidity).
+
+In this document we describe how some of the gaps between the high-level source
+and the low-level bytecode can affect our understanding of the Prover's outputs,
+and recommended solutions.
+
+## Loops
+
+### Determining the number of needed iterations
+
+The Prover deals with loops by unrolling them a constant number of times 
+(see {ref}`--loop_iter`).
+Furthermore, it can add an assertion that the number of unrolls applies
+was sufficient to fully capture all of the loop's behavior, which is usually useful
+in loops that are known to have a constant number of iterations.
+Otherwise, the user can opt-in to assume the unroll bound was sufficient
+(see {ref}`--optimistic_loop`).
+
```

Comment:
Maybe we should add an example in pseudo-code of loop unrolling? 
It would also help understand how many times the exit condition is evaluated.

Reply:
the pseudo code of the unroller itself? this seems a bit of a stretch, but I can show the 'unrolled' Solidity example